source/imaer-shared/src/main/java/nl/overheid/aerius/shared/emissions/FarmlandEmissionsCalculator.java

```diff
@@ -22,51 +22,55 @@
-      }
+    for (final AbstractFarmlandActivity activity : farmlandSource.getSubSources()) {
+      activity.accept(this, summed);
     }
 
     final Map<Substance, Double> result = new EnumMap<>(Substance.class);
     summed.forEach((key, value) -> result.put(key, value.doubleValue()));
     return result;
   }
 
-  private void updateEmissions(final Map<Substance, Double> emissions, final Map<Substance, BigDecimal> summedEmissions) {
-    emissions.forEach(
-        (key, value) -> summedEmissions.merge(key, BigDecimal.valueOf(value), (v1, v2) -> v1.add(v2)));
+  @Override
+  public void visit(final CustomFarmlandActivity activity, final Map<Substance, BigDecimal> summedEmissions) {
+    activity.getEmissions().forEach(
+        (key, value) -> summedEmissions.merge(key, BigDecimal.valueOf(value), BigDecimal::add));
   }
 
-  private void updateEmissions(final FarmlandGrazingActivity activity, final Map<Substance, BigDecimal> summedEmissions) {
-    Map<Substance, Double> repositoryEmissions = farmlandEmissionFactorSupplier.getGrazingEmissionFactors(activity.getGrazingCategoryCode());
-    Map<Substance, Double> userEmissions = activity.getEmissions();
-
-    // Prefer emission factors supplied by the farmlandEmissionFactorSupplier
-    Map<Substance, Double> activityEmissionFactors = repositoryEmissions.isEmpty() ? userEmissions : repositoryEmissions;
+  @Override
+  public void visit(final StandardFarmlandActivity activity, final Map<Substance, BigDecimal> summedEmissions) throws AeriusException {
+    final Map<Substance, Double> emissionFactors = farmlandEmissionFactorSupplier
+        .getFarmSourceEmissionFactors(activity.getFarmSourceCategoryCode());
 
-    if (activity.getFarmEmissionFactorType() == FarmEmissionFactorType.PER_ANIMAL_PER_DAY) {
-      activityEmissionFactors.forEach((key, value) ->
-          summedEmissions.merge(key, BigDecimal.valueOf(value).multiply(BigDecimal.valueOf(activity.getDays())), (v1, v2) -> v1.add(v2))
+    final FarmEmissionFactorType emissionFactorType = farmlandEmissionFactorSupplier.getFarmEmissionFactorType(activity.getFarmSourceCategoryCode());
+    if (emissionFactorType == FarmEmissionFactorType.PER_ANIMAL_PER_DAY) {
+      emissionFactors.forEach((key, value) ->
+          summedEmissions.merge(key, BigDecimal.valueOf(value).multiply(BigDecimal.valueOf(activity.getNumberOfDays()))
+              .multiply(BigDecimal.valueOf(activity.getNumberOfAnimals())), BigDecimal::add)
+      );
+    } else if (emissionFactorType == FarmEmissionFactorType.PER_ANIMAL_PER_YEAR) {
+      emissionFactors.forEach((key, value) ->
+          summedEmissions.merge(key, BigDecimal.valueOf(value).multiply(BigDecimal.valueOf(activity.getNumberOfAnimals())), BigDecimal::add)
       );
     } else {
-      updateEmissions(activityEmissionFactors, summedEmissions);
+      throw new AeriusException(ImaerExceptionReason.INTERNAL_ERROR, "Invalid farm emission factor type for StandardFarmlandActivity.");
```

Comment:
Was intended to be `GML_FARM_EMISSION_FACTOR_TYPE_UNKNOWN` instead of the internal error I guess (else that one was added for nothing). 

Come to think of it, that would be more an indication that the `farmSourceCategoryCode` was unknown in the first place if it reaches a user, so it should perhaps be captured by a validation part (but can do that later on with the actual IMAER changes), and then an internal error might make sense here: It would only reach this in case of a new type that we didn't implement fully yet, or if nothing was validated.

Reply:
Right, looking at other EmissionCalculators throwing an internal error in this case is the expected behavior. I'll remove `GML_FARM_EMISSION_FACTOR_TYPE_UNKNOWN`